### PR TITLE
Azure sentinel assignement options

### PR DESCRIPTION
**Pull request by asieberle**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
- Added support for incident assignment based on *AssigneeObjectID*  to ***azure-sentinel-update-incident*** command.
- Added support for *unassign* incidents to ***azure-sentinel-update-incident*** command.
- Added the context keys AzureSentinel.Incident.AssigneeObjectID, AzureSentinel.Incident.AssigneeUPN.

## Must have
- [ ] Tests
- [x] Documentation

### CODE REVIEW

**Comment by asieberle** [comment]
don´t know why circle ci complains about release notes file,  _demisto-sdk doc-review_ shows no issues

**Comment by melamedbn** [comment]
Hi @asieberle,

Thanks for the work and time put in the contribution!
I saw many files that were added to the PR with changes that should be reverted, please remove the following file types from the PR to make the review process more efficient:
1. Classifier
2. Incident type
3. Incident fields
4. Layout

Best regards,
Ben

File: Packs/AzureSentinel/TestPlaybooks/TestAzureSentinelPlaybookV2.yml [hunk 1]
```diff
@@ -1991,4 +1991,4 @@ inputs:
 outputs: []
 sourceplaybookid: TestAzureSentinelPlaybook
 fromversion: 5.0.0
-description: Testing Azure Sentinel integration
+description: Testing Azure Sentinel integration.
```
Comment (MLainer1) [inline Packs/AzureSentinel/TestPlaybooks/TestAzureSentinelPlaybookV2.yml:1994]: Please revert

File: Packs/AzureSentinel/TestPlaybooks/NonCircleTests/playbook-TestAzureSentinelPlaybook.yml [hunk 2]
```diff
@@ -13,6 +13,7 @@ tasks:
       name: ""
       iscommand: false
       brand: ""
+      description: ''
```
Comment (MLainer1) [inline Packs/AzureSentinel/TestPlaybooks/NonCircleTests/playbook-TestAzureSentinelPlaybook.yml:16]: Please revert the changes in this playbook

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 3]
```diff
@@ -545,13 +551,22 @@ script:
...
     - description: Incident labels. Note that all labels will be set as labelType='User'.
-      isArray: true
       name: labels
+      isArray: true
     - description: The subscription ID.
       name: subscription_id
     - description: The resource group name.
       name: resource_group_name
+    - auto: PREDEFINED
+      description: If set to true the assignee will be removed and incident will remain unassigned.
```
Comment (MLainer1) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:564]:
```suggestion
      description: If set to true the assignee will be removed and the incident will remain unassigned. Overrides the other arguments related to the assignee.
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py [hunk 4]
```diff
@@ -360,10 +364,15 @@ def get_update_incident_request_data(client: AzureSentinelClient, args: Dict[str
...
-        owner['email'] = assignee_email
+    if assignee_objectid:
+        owner = {'objectId': assignee_objectid}
+    else:
+        if user_principal_name:
+            owner = {'userPrincipalName': user_principal_name}
+        if assignee_email:
+            owner['email'] = assignee_email
+    if unassign == 'true':
+        owner = {}
```
Comment (MLainer1) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py:375]:
```suggestion
    if unassign == 'true':
        owner = {}
    elif assignee_objectid:
        owner = {'objectId': assignee_objectid}
    else:
        if user_principal_name:
            owner = {'userPrincipalName': user_principal_name}
        if assignee_email:
            owner['email'] = assignee_email
```
This order makes better understanding of the hierarchy of the arguments

**Review by MLainer1 — CHANGES_REQUESTED** [review]
Nice work!, please go over the comments, and let me know if you need any assistance

**Comment by asieberle** [comment]
@melamedbn 
Just want to mention that changes were made by demisto sdk, not a manual edit. So when the next developer will use the demisto-sdk format command, those changes will appear again!?

File: Packs/AzureSentinel/TestPlaybooks/NonCircleTests/playbook-TestAzureSentinelPlaybook.yml (hunk 2 above)
Comment (asieberle) [reply Packs/AzureSentinel/TestPlaybooks/NonCircleTests/playbook-TestAzureSentinelPlaybook.yml]: done - want to mention that changes were made by demisto sdk, not a manual edit

**Comment by MLainer1** [comment]
Hi @asieberle, can you revert the changes made to the .json files in the PR? it looks like the changes were not made by you

**Comment by asieberle** [comment]
I reverted the json changes made by demisto-sdk

**Comment by MLainer1** [comment]
@asieberle Thank you, now just the conflicts are blocking the merging of the PR, if you can handle them that will be great :)

**Comment by asieberle** [comment]
@MLainer1 , hope I´ve done in correctly

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 5]
```diff
@@ -545,13 +551,22 @@ script:
       name: assignee_email
     - description: The user principal name of the client. Note that the updated API field is `owner.userPrincipalName`.
       name: user_principal_name
+    - description: Object ID of assignee, this is the unique objectid of a user to update assignee of incident Updated key is owner.objectId
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:554]:
```suggestion
    - description: Object ID of assignee. This is the unique object ID of a user to update the assignee of the incident. Updated key is owner.objectId.
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 6]
```diff
@@ -545,13 +551,22 @@ script:
...
     - description: Incident labels. Note that all labels will be set as labelType='User'.
-      isArray: true
       name: labels
+      isArray: true
     - description: The subscription ID.
       name: subscription_id
     - description: The resource group name.
       name: resource_group_name
+    - auto: PREDEFINED
+      description: If set to true the assignee will be removed and the incident will remain unassigned. Overrides the other arguments related to the assignee.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:564]:
```suggestion
      description: If set to true, the assignee will be removed and the incident will remain unassigned. Overrides the other arguments related to the assignee.
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 7]
```diff
@@ -576,6 +591,12 @@ script:
     - contextPath: AzureSentinel.Incident.AssigneeEmail
       description: The email address of the incident assignee.
       type: String
+    - contextPath: AzureSentinel.Incident.AssigneeObjectID
+      description: The objectId of the incident assignee.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:595]:
```suggestion
      description: The object ID of the incident assignee.
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml [hunk 8]
```diff
@@ -576,6 +591,12 @@ script:
...
+      type: string
+    - contextPath: AzureSentinel.Incident.AssigneeUPN
+      description: Userprincipalname of incident assignee.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.yml:598]:
```suggestion
      description: The user principal name of the incident assignee.
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel_description.md [hunk 9]
```diff
@@ -37,4 +37,4 @@ If you have a dedicated server URL, enter it in the *Server Url* parameter.
 
 ## Get the Additional Instance Parameters
 
-To get the *Subscription ID*, *Workspace Name* and *Resource Group* parameters, in the Azure Portal navigate  to **Azure Sentinel** > your workspace > **Settings** and click the **Workspace Settings** tab.
+To get the *Subscription ID*, *Workspace Name* and *Resource Group* parameters, in the Azure Portal navigate  to **Azure Sentinel** > your workspace > **Settings** and click the **Workspace Settings** tab.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel_description.md:40]:
```suggestion
To get the *Subscription ID*, *Workspace Name*, and *Resource Group* parameters, in the Azure Portal navigate to **Azure Sentinel** > your workspace > **Settings** and click the **Workspace Settings** tab.
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/README.md [hunk 10]
```diff
@@ -125,6 +125,8 @@ Gets a single incident from Azure Sentinel.
 | AzureSentinel.Incident.Status | String | The incident status. | 
 | AzureSentinel.Incident.AssigneeName | String | The name of the incident assignee. | 
 | AzureSentinel.Incident.AssigneeEmail | String | The email address of the incident assignee. | 
+| AzureSentinel.Incident.AssigneeObjectID | String | The objectId of the incident assignee. | 
+| AzureSentinel.Incident.AssigneeUPN | String | The userprincipalname of the incident assignee. |
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/README.md:129]:
```suggestion
| AzureSentinel.Incident.AssigneeObjectID | String | The object ID of the incident assignee. | 
| AzureSentinel.Incident.AssigneeUPN | String | The user principal name of the incident assignee. | 
```

File: Packs/AzureSentinel/Integrations/AzureSentinel/README.md [hunk 11]
```diff
@@ -716,6 +718,8 @@ Updates a single incident in Azure Sentinel.
...
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/Integrations/AzureSentinel/README.md:722]:
```suggestion
| AzureSentinel.Incident.AssigneeObjectID | String | The object ID of the incident assignee. | 
| AzureSentinel.Incident.AssigneeUPN | String | The user principal name of the incident assignee. | 
```

File: Packs/AzureSentinel/ReleaseNotes/1_5_33.md [hunk 12]
```diff
@@ -0,0 +1,8 @@
+
+#### Integrations
+
+##### Microsoft Sentinel
+- Added support for incident assignment based on *AssigneeObjectID*  to ***azure-sentinel-update-incident*** command.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/ReleaseNotes/1_5_33.md:5]:
```suggestion
- Added support for incident assignment based on *AssigneeObjectID*  in the ***azure-sentinel-update-incident*** command.
```

File: Packs/AzureSentinel/ReleaseNotes/1_5_33.md [hunk 13]
```diff
@@ -0,0 +1,8 @@
...
+- Added support for *unassign* incidents to ***azure-sentinel-update-incident*** command.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/ReleaseNotes/1_5_33.md:6]:
```suggestion
- Added support for *unassign* incidents in the ***azure-sentinel-update-incident*** command.
```

File: Packs/AzureSentinel/ReleaseNotes/1_5_33.md [hunk 14]
```diff
@@ -0,0 +1,8 @@
...
+- Added the context keys AzureSentinel.Incident.AssigneeObjectID, AzureSentinel.Incident.AssigneeUPN.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/ReleaseNotes/1_5_33.md:7]:
```suggestion
- Added the following context keys 
   - *AzureSentinel.Incident.AssigneeObjectID*
   - *AzureSentinel.Incident.AssigneeUPN*
```

File: Packs/AzureSentinel/ReleaseNotes/1_5_33.md [hunk 15]
```diff
@@ -0,0 +1,8 @@
...
+- Updated the Docker image to: demisto/python3:3.10.13.80014.
```
Comment (ShirleyDenkberg) [inline Packs/AzureSentinel/ReleaseNotes/1_5_33.md:8]:
```suggestion
- Updated the Docker image to: *demisto/python3:3.10.13.80014*.
```

**Comment by asieberle** [comment]
@MLainer1 its unclear to me if I should commit the suggested documentation changes by ShirleyDenkberg or if this is addressed to you. I´m ok with all that changes, so you can commit them or I can do it later tomorrow

**Comment by MLainer1** [comment]
> @MLainer1 its unclear to me if I should commit the suggested documentation changes by ShirleyDenkberg or if this is addressed to you. I´m ok with all that changes, so you can commit them or I can do it later tomorrow

it does not really matter which one of us will commit those changes, feel free to commit them if you see that I did not do it

**Comment by MLainer1** [comment]
Nice work @asieberle  ! is the PR ready for merge? everything looks good from my side

**Review by MLainer1 — APPROVED** [review]

**Comment by asieberle** [comment]
@MLainer1 from my pov this PR can be merged
